### PR TITLE
Add command usage

### PR DESCRIPTION
**Pull request by Learath2**
I think this makes console more friendly.

The patch was originally made by @H-M-H for ddnet but it wasn't really convenient to cherry-pick.

### CODE REVIEW

**Comment by fokkonaut** [comment]
You forgot to add this for the command system

**Comment by Learath2** [comment]
They both use the same argument parser, I just didn't add usage to the
commands because they have their own way of handling it and I didn't
exactly figure out how best to do it there.

**Comment by fokkonaut** [comment]
It should be done though

**Comment by Learath2** [comment]
@fokkonaut Take a look now pls

**Comment by fokkonaut** [comment]
Looks good already, but I'd like to send argformat from the server, so the client can display e.g. `login s[username] s[password]` or `map r[mapname]` or `points ?r[name]`

**Comment by Learath2** [comment]
That's what this does

**Comment by fokkonaut** [comment]
I thought this would write <float> or <number> or <string> instead of e.g. i[id]

**Comment by Learath2** [comment]
It writes float number or string if you don't have a description

**Comment by fokkonaut** [comment]
LGTM then.

File: src/game/client/components/chat.cpp [hunk 1]
```diff
@@ -1415,34 +1415,58 @@ void CChat::HandleCommands(float x, float y, float w)
...
+							str_format(aBuf, sizeof(aBuf), "%cfloat%c",
+								Brackets[Optional][0], Brackets[Optional][1]);
+							break;
+						case 'r':
+						case 's':
+							str_format(aBuf, sizeof(aBuf), "%cstring%c",
+								Brackets[Optional][0], Brackets[Optional][1]);
+							break;
+						default:
+							goto illformed;
```
Comment (oy) [inline src/game/client/components/chat.cpp]: could you remove that goto

**Comment by Learath2** [comment]
I could replace the entire thing with an if/else chain, but is that really
necessary?

**Comment by fokkonaut** [comment]
You could make a var Found = true, which gets set false on default:, then you can execute the rest only when Found is false

**Comment by oy** [comment]
We don't have a single goto statement in the teeworlds code and should stick to it.
It looks safe here, but people can do bad things with it.

**Pull request merged**